app/models/detailed_guide.rb

```diff
@@ -62,6 +65,35 @@ def related_detailed_guide_ids=(detailed_guide_ids)
     self.related_documents = other_related_documents + detailed_guide_documents
   end
 
+  def need_ids
+    return @need_ids if @need_ids
+    return [] unless document # TODO: Remove
+
+    response = Whitehall.publishing_api_v2_client.get_links(
+      document.content_id
+    )
+
+    return unless response
+
+    @need_ids = response["links"]["meets_user_needs"]
+  end
+
+  def patch_meets_user_needs_links
+    return unless @need_ids
```

Comment:
Is this being caused by the schema issue in my other comment? Either way, while you've still got a database field for `need_ids` this might be difficult to do.